### PR TITLE
test: Faster Integration Testing

### PR DESCRIPTION
**Pull request by petervdmeer**
**Feature:** This PR improves the speed of integration testing by removing limits and reducing overhead.

## Feature Details
Removed the 60 steps-per-second limit during testing and reduced the 3D/graphics drawing to once every 30 frames during testing. The 60 steps-per-second is valuable for players that want to run the game at a speed which humans can interact with, but for testing we are better off running as fast as the system can run for all code that is unaware of time in the real world (which is nearly all of the game-code).
Combined some small tests into a single larger test to reduce overhead.

## UI Screenshots
N/A

## Usage Examples
Just run the tests.

## Testing Done
```
time ./tests/run_tests.sh ./endless-sky ./
TAP version 13
# ***********************************************
# ***         ES Autotest-runner              ***
# ***********************************************
# Using Test Anything Protocol for reporting test-results
1..4
# Running test "Test-Framework - Calls - Arithmetic - Loops - Keyboard Control":
ok 1 "Test-Framework - Calls - Arithmetic - Loops - Keyboard Control"
# Running test "Test-Framework - Navigate":
ok 2 "Test-Framework - Navigate"
# Running test "Tests - Ember Wastes Navigation":
ok 3 "Tests - Ember Wastes Navigation"
# Running test "Tests - Wormhole Navigation":
ok 4 "Tests - Wormhole Navigation"

# tests 4
# pass 4

real	0m18.455s
user	0m31.469s
sys	0m4.354s
```

## Performance Impact
Impact on CI duration and CI resource usage is expected to be positive.

### CODE REVIEW

File: data/tests/tests_framework.txt [hunk 1]
```diff
@@ -9,20 +9,23 @@
 # PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 
 test "Test-Framework - Empty Testcase"
-	status "active"
+	# Partial; saving test-overhead by only running this test in combination with other test.
+	status partial
 	description "Test with no test sequence. Only starts and exits the game."
```
Comment (tehhowch) [inline data/tests/tests_framework.txt:14]: I don't feel making these changes is necessary.

What we could do instead is move them into a plugin and include the plugin in the CI stage whenever framework code is modified.

File: source/main.cpp [hunk 2]
```diff
@@ -314,7 +319,8 @@ void GameLoop(PlayerInfo &player, const Conversation &conversation, const string
 		
 		GameWindow::Step();
 		
-		timer.Wait();
+		if(testContext.testToRun.empty())
```
Comment (tehhowch) [inline source/main.cpp:322]: Please include a comment, e.g.
```c++
// Run automated tests as quickly as possible.
```

Comment (petervdmeer) [reply source/main.cpp]: Good idea. Added a comment about this (and clarified a related comment).

File: data/tests/tests_framework.txt (hunk 1 above)
Comment (petervdmeer) [reply data/tests/tests_framework.txt]: Yes, we could  move tests to their own plugins; I'll keep that for a follow-up PR. I restored the tests to their previous status in this PR.

**Pull request merged**